From 802976bb4039e9adfd6c3f80816a78a305a545ab Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Mon, 30 Jul 2018 14:52:09 +0800
Subject: [PATCH] tweak product short name

The yocto named its production with braces which caused short name
with illegal character that broke lvm partition.

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>

Rebase for anaconda 39 on 20240304.

Signed-off-by: Kai Kang <kai.kang@windriver.com>

Rebase for anaconda 42 on 20250328.

Signed-off-by: Kai Kang <kai.kang@windriver.com>
---
 pyanaconda/core/product.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/pyanaconda/core/product.py b/pyanaconda/core/product.py
index 397c1b1f03..77c5a9e83c 100644
--- a/pyanaconda/core/product.py
+++ b/pyanaconda/core/product.py
@@ -56,7 +56,9 @@ def shorten_product_name(long_name):
     """
     product_short_name = long_name.lower()  # pylint: disable=no-member
 
-    if long_name.count(" "):  # pylint: disable=no-member
+    if '(' in product_short_name and ')' in product_short_name:
+        product_short_name = product_short_name.split()[0]
+    elif long_name.count(" "):  # pylint: disable=no-member
         product_short_name = ''.join(s[0] for s in product_short_name.split())
 
     return product_short_name
